gitlab/client.py

```diff
@@ -966,17 +966,23 @@ def per_page(self) -> int:
         return int(self._per_page)
 
     @property
-    def total_pages(self) -> int:
+    def total_pages(self) -> Optional[int]:
         """The total number of pages."""
-        if TYPE_CHECKING:
-            assert self._total_pages is not None
+        if self._total_pages is None:
+            # NOTE(jlvillal): When a query returns more than 10,000 items, GitLab
+            # doesn't return the header for this value. So we return None.
+            # https://docs.gitlab.com/ee/user/gitlab_com/index.html#pagination-response-headers
+            return None
         return int(self._total_pages)
```

Comment:
If the header being returned is the more common scenario, should we be inverting this into try/except?

```suggestion
        try:
            return int(self._total_pages)
        except TypeError:
            # NOTE(jlvillal): When a query returns more than 10,000 items, GitLab
            # doesn't return the header for this value. So we return None.
            # https://docs.gitlab.com/ee/user/gitlab_com/index.html#pagination-response-headers
            return None
```

Reply:
Not sure why but I'm not a fan of try/except for standard use cases. I'll switch it to a 'if blah is not None`.